src/account-info/AccountDetails.js

```diff
@@ -43,13 +43,16 @@ class AccountDetails extends PureComponent<Props> {
   render() {
     const { realm, user, userStatusText } = this.props;
 
-    // The set of timezone names in the tz database is subject to change over
-    // time. Handle unrecognized timezones by quietly discarding them.
-    let localTime: string | null;
-    try {
-      localTime = `${nowInTimeZone(user.timezone)} Local time`;
-    } catch (err) {
-      localTime = null;
+    let localTime: string | null = null;
+    // `user` will be '' or undefined if it is a bot (see comment at src/api/modelTypes);
+    // `user` will be '' if a human user with no timezone set.
```

Comment:
This sort of knowledge is great to write down at the type definition in `modelTypes.js` -- that way it's there for when we refer to this property from some other code, unrelated to this file.

(Also, per my previous comment interpreting what Tim said, I think the story is a bit more homogeneous than this: it may be `''` or missing/undefined if there's no timezone set, which in particular is true of all bots. The "missing" case is the past for cross-realm bots, and a possible future for... I think probably for all types of users.)

Reply:
> it may be `''` or missing/undefined if there's no timezone set, which in particular is true of all bots.

Hmm, are you sure about the first part? I don't think it follows from Tim's [comment](https://github.com/zulip/zulip-mobile/pull/3789#issuecomment-581218576) that human users will ever have it missing/undefined. But I do see that bots may have it missing/undefined, and that all `CrossRealmBot`s, and some, `User`s are bots (the single-realm bots); thanks for explaining the subtlety there and for https://github.com/zulip/zulip-mobile/commit/60807bde20d7e6a43db17d30b75c14f56c540676. 🙂

If I'm understanding correctly, then, these comments by chance ended up being conceptually correct, but I bet it would be more practical to refer to our type aliases, by name, right, so something like this:

```
// `user.timezone` will be '' or undefined for CrossRealmBots (see CrossRealmBot comment at src/api/modelTypes);
// `user.timezone` will be '' for Users, but in future may be '' or undefined (see User comment at modelTypes).
```

Reply:
> I don't think it follows from Tim's [comment](https://github.com/zulip/zulip-mobile/pull/3789#issuecomment-581218576) that human users will ever have it missing/undefined.

Hmm, yes, on rereading the comment I agree with that reading.

Still, I expect it's possible for human users to have no meaningful timezone set -- just from an onboarding UX perspective. And spot-checking the right sidebar on czo, I found a user with no "local time" line in their profile popup, which seems likely to mean that.

Then searching through the server's `zerver/models.py` for the timezone field on users, here it is with a nice comment:
```
    # A timezone name from the `tzdata` database, as found in pytz.all_timezones.
    #
    # The longest existing name is 32 characters long, so max_length=40 seems
    # like a safe choice.
    #
    # In Django, the convention is to use an empty string instead of NULL/None
    # for text-based fields. For more information, see
    # https://docs.djangoproject.com/en/1.10/ref/models/fields/#django.db.models.Field.null.
    timezone = models.CharField(max_length=40, default=u'')  # type: str
```
So "no known timezone" is represented as `''` inside the server. Which is probably why it's the same way, for now, in the API.

And although Tim's comment about the future only mentioned bots, if we were to make that change I don't see why it'd be conditioned on being a bot or not -- suppressing `timezone: '',` would have the same size benefits for human users as for bots. So as a predictive matter, I think it's likely we'd want to do it for human users the same as for bots. In any case when we follow Tim's actionable advice on the matter:

> I'd recommend writing the mobile app in a way that treats the value being missing the same as `''`.

the easiest way to do that is to apply it the same way for all users, and happily that's also the most broadly future-proof way.

Reply:
> but I bet it would be more practical to refer to our type aliases, by name, right

Yep -- I was just about to add a small comment about this, and then saw you'd already mentioned it here :wink: